src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -542,74 +530,105 @@ class CVulkanLogicalDevice final : public ILogicalDevice
+                        infoDst->range = infoSrc->info.buffer.size;
                     }
-                    else
+                }
+            } break;
+
+            case asset::IDescriptor::E_CATEGORY::EC_IMAGE:
+            {
+                vk_writeDescriptorSets[i].pImageInfo = reinterpret_cast<size_t>(vk_writeDescriptorSets[i].pImageInfo) + vk_imageInfos.data();
+
+                const auto* infoSrc = pDescriptorWrites[i].info;
+                auto* infoDst = const_cast<VkDescriptorImageInfo*>(vk_writeDescriptorSets[i].pImageInfo);
+                for (uint32_t j = 0; j < pDescriptorWrites[i].count; ++j, ++infoSrc, ++infoDst)
+                {
+                    if (infoSrc->info.image.imageLayout != asset::IImage::E_LAYOUT::EL_UNDEFINED)
                     {
-                        vk_bufferViews[bufferViewOffset + j] = dummyBufferView;
+                        VkSampler vk_sampler = VK_NULL_HANDLE;
+                        if (infoSrc->info.image.sampler && (infoSrc->info.image.sampler->getAPIType() == EAT_VULKAN))
+                            vk_sampler = static_cast<const CVulkanSampler*>(infoSrc->info.image.sampler.get())->getInternalObject();
+
+                        infoDst->sampler = vk_sampler;
+                        infoDst->imageView = static_cast<const CVulkanImageView*>(infoSrc->desc.get())->getInternalObject();
+                        infoDst->imageLayout = static_cast<VkImageLayout>(infoSrc->info.image.imageLayout);
                     }
                 }
-
-                vk_writeDescriptorSets[i].pTexelBufferView = vk_bufferViews.data() + bufferViewOffset;
-                bufferViewOffset += pDescriptorWrites[i].count;
             } break;
 
-            case asset::IDescriptor::EC_ACCELERATION_STRUCTURE:
+            case asset::IDescriptor::E_CATEGORY::EC_BUFFER_VIEW:
             {
-                // Get WriteAS
-                auto& writeAS = vk_writeDescriptorSetAS[i];
-                writeAS = { VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET_ACCELERATION_STRUCTURE_KHR, nullptr };
-                // Fill Write AS
-                for (uint32_t j = 0u; j < pDescriptorWrites[i].count; ++j)
+                vk_writeDescriptorSets[i].pTexelBufferView = reinterpret_cast<size_t>(vk_writeDescriptorSets[i].pTexelBufferView) + vk_bufferViews.data();
+
+                const auto* infoSrc = pDescriptorWrites[i].info;
+                auto* infoDst = const_cast<VkBufferView*>(vk_writeDescriptorSets[i].pTexelBufferView);
+                for (uint32_t j = 0u; j < pDescriptorWrites[i].count; ++j, ++infoSrc, ++infoDst)
                 {
-                    VkAccelerationStructureKHR vk_accelerationStructure = static_cast<const CVulkanAccelerationStructure*>(pDescriptorWrites[i].info[j].desc.get())->getInternalObject();
-                    vk_accelerationStructures[j + accelerationStructuresOffset] = vk_accelerationStructure;
+                    if (infoSrc->info.buffer.size)
```

Comment:
why do you check this?

Reply:
Same as https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#discussion_r1124149562

Reply:
ok, then same as https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#discussion_r1127710135

Reply:
Resolved.